### PR TITLE
drivers: crypto: hisilicon: add authenc algorithm

### PR DESCRIPTION
**Pull request by LeiSen62**
Add authenc algorithm for hisilicon SEC driver

### CODE REVIEW

File: core/drivers/crypto/hisilicon/sec_authenc.c [hunk 1]
```diff
@@ -0,0 +1,824 @@
...
+
+	if (!dinit->nonce.data || !dinit->nonce.length) {
+		EMSG("iv is NULL");
+		return TEE_ERROR_BAD_PARAMETERS;
+	}
+
+	return TEE_SUCCESS;
+}
+
+static bool is_hw_support(struct drvcrypt_authenc_init *dinit)
```
Comment (jenswi-linaro) [inline core/drivers/crypto/hisilicon/sec_authenc.c:293]: How about `is_hw_supported()`?

File: core/drivers/crypto/hisilicon/sec_authenc.c [hunk 2]
```diff
@@ -0,0 +1,824 @@
...
+{
+	struct authenc_ctx *ae_drv_ctx = NULL;
+	TEE_Result ret = TEE_SUCCESS;
+	bool hw_support = false;
+
+	ret = authenc_init_params_check(dinit);
+	if (ret)
+		return ret;
+
+	ae_drv_ctx = (struct authenc_ctx *)dinit->ctx;
```
Comment (jenswi-linaro) [inline core/drivers/crypto/hisilicon/sec_authenc.c:495]: There's no need to cast a void pointer, if there are other places like this please fix those too.

File: core/drivers/crypto/hisilicon/sec_authenc.c [hunk 3]
```diff
@@ -0,0 +1,824 @@
...
+
+	ae_drv_ctx->civ[0] = cl;
+	ae_drv_ctx->civ[MAX_IV_SIZE - 1] = IV_CTR_INIT;
+
+	return TEE_SUCCESS;
+}
+
+static TEE_Result sec_aead_get_dma(struct authenc_ctx *ae_drv_ctx)
+{
+	ae_drv_ctx->key_dma = virt_to_phys(ae_drv_ctx->key);
```
[the remark this answers is not in the document]
Comment (LeiSen62) [reply core/drivers/crypto/hisilicon/sec_authenc.c:411]: I don't think virt_to_phys will fail here, how about removing the return value checking？

[the remark this answers is not in the document]
Comment (jenswi-linaro) [reply core/drivers/crypto/hisilicon/sec_authenc.c:411]: OK

**Review by jenswi-linaro — COMMENTED** [review]
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by LeiSen62** [comment]
@jforissier All comments have been solved.

**Pull request merged**